main/streams/streams.c

```diff
@@ -670,12 +674,19 @@ PHPAPI zend_result _php_stream_fill_read_buffer(php_stream *stream, size_t size)
 					stream->readbuflen - stream->writepos
 					);
 			if (justread < 0) {
-				return FAILURE;
+				retval = FAILURE;
+				goto out;
 			}
 			stream->writepos += justread;
 		}
-		return SUCCESS;
+		retval = SUCCESS;
+	}
+
+out:
+	if (old_eof != stream->eof) {
+		php_stream_notify_completed(PHP_STREAM_CONTEXT(stream));
```

Comment:
I don't think we need to check this every time. If it's already in the buffer, then this is pointless. I would just move it to the block where it's needed.